### PR TITLE
Display error properties when tailing logs

### PR DESCRIPTION
**Pull request by mimi-stripe**
[wip] 

### Reviewers
r? @ob-stripe 
cc @stripe/dev-platform

 ### Summary
Changes to tailer to allow 'logs tail' to accept error info from request log events, and print to user.

### CODE REVIEW

File: pkg/logtailing/tailer.go [hunk 1]
```diff
@@ -74,6 +74,17 @@ type EventPayload struct {
 	RequestID string `json:"request_id"`
 	Status    int    `json:"status"`
 	URL       string `json:"url"`
+	Error     string `json:"error"`
```
Comment (ob-stripe) [inline pkg/logtailing/tailer.go]: I think this should be typed as `RedactedError` instead of `string`.

File: pkg/logtailing/tailer.go [hunk 2]
```diff
@@ -74,6 +74,17 @@ type EventPayload struct {
...
+}
+
+// RedactedError is the mapping for fields in error from an EventPayload
+type RedactedError struct {
```
Comment (ob-stripe) [inline pkg/logtailing/tailer.go]: We can probably just call this `RequestLogError` or something. From the CLI's perspective it's not really important that the error is "redacted".

File: pkg/logtailing/tailer.go [hunk 3]
```diff
@@ -190,6 +201,15 @@ func (tailer *Tailer) processRequestLogEvent(msg websocket.IncomingMessage) {
 	color := ansi.Color(os.Stdout)
 	outputStr := fmt.Sprintf("%s [%d] %s %s [%s]", color.Faint(localTime), coloredStatus, payload.Method, payload.URL, requestLink)
 	fmt.Println(outputStr)
+
+	var payloadError RedactedError
+	if err := json.Unmarshal([]byte(payload.Error), &payloadError); err != nil {
```
Comment (ob-stripe) [inline pkg/logtailing/tailer.go]: Assuming the stripecli service sends the `error` field as a hash, not as a string, we don't need to unmarshal it ourselves. Just unmarshaling the top-level `EventPayload` will do the right thing.

Comment (mimi-stripe) [reply pkg/logtailing/tailer.go]: Oh cool, I didn't know it would work recursively like that, nice!

**Pull request merged**